SOLID_No1.php

```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<html>
+<body>
+  <?php
+
+  interface AreaInterface{
+    public  function calculateArea();
+  }
+
+  class Rectangle implements AreaInterface{
+    protected $width;
+    protected $height;
+
+    public function setWidth($width)
+    {
+        $this->width = $width;
+    }
+
+    public function setHeight($height)
+    {
+        $this->height = $height;
+    }
+
+    public  function calculateArea(){
+      $area = $this->width * $this->height;
+      return $area;
+    }
+}
+
+class Square extends Rectangle
```

Comment:
Please checking in here. Square it violated Liskov Substitution Principle.
Because Square only need 1 property width instead of 2 properties width and height.
Please removing extends Rectangle.

Reply:
Fixed. Please check again.